ironmon_tracker/screens/TrackerScreen.lua

```diff
@@ -802,8 +802,17 @@ function TrackerScreen.drawPokemonInfoArea(data)
 				end
 			else
 				evoTextColor = Theme.COLORS["Default text"]
+			end			
+			-- Adding in a frienship loop lookup to try and print out multicolor letters based on freindship :) -Tainted_wolf
+			if (data.p.evo == PokemonData.Evolutions.FRIEND) and Options["Determine friendship readiness"] and Tracker.Data.isViewingOwn then 
+				local evoFriendshipRatio = ((data.p.friendship-data.p.friendshipBase)/(Program.GameData.friendshipRequired-data.p.friendshipBase)) -- Read in friendship level from the pokemon and get ratio
+				local evoTypeStringLength = string.len(abbreviationText) --setting up so we know how many letters to color
+				local numberLettersToHighlight =  math.floor((evoTypeStringLength * evoFriendshipRatio))
+				local highlightedString = string.sub(abbreviationText , 1, numberLettersToHighlight)
```

Comment:
Some formatting cleanup, if you wouldn't mind:

1. With some spacing and less parens/comments:
```lua
-- Before
local evoFriendshipRatio = ((data.p.friendship-data.p.friendshipBase)/(Program.GameData.friendshipRequired-data.p.friendshipBase)) -- Read in friendship level from the pokemon and get ratio
```
```lua
-- After
local percentFill = (data.p.friendship - data.p.friendshipBase) / (Program.GameData.friendshipRequired - data.p.friendshipBase)
```

2. The variable `evoTypeStringLength` is only used once, and the name is as long as the value. You can remove this variable entirely

3. The `math.floor()` has an extra set of parens.

4. The line of code `string.sub(abbreviationText , 1, numberLettersToHighlight)` can become `abbreviationText:sub(1, numberLettersToHighlight)`. Notably, just a tad shorter and fixes the location of the spaces.

Reply:
Yeah I tend to just slam everything together since that is how my day job does it. I missed this one and will try to keep anything else I tackle a little cleaner 👍